File: src/java/VRE.Vridge.API/VRE.Vridge.API.Client/src/main/java/com/riftcat/vridge/api/client/java/control/responses/ControlResponseHeader.java
package com.riftcat.vridge.api.client.java.control.responses;

import com.riftcat.vridge.api.client.java.control.BaseControlMessage;
import com.riftcat.vridge.api.client.java.control.ControlResponseCode;

public class ControlResponseHeader extends BaseControlMessage {

    // Predefined responses
    public static ControlResponseHeader ResponseInUse;
    public static ControlResponseHeader ResponseClientOutdated;
    public static ControlResponseHeader ResponseNotAvailable;

    static{
        // Predefined responses
        ResponseNotAvailable  = new ControlResponseHeader();
        ResponseNotAvailable.Code = ControlResponseCode.NotAvailable;

        ResponseClientOutdated  = new ControlResponseHeader();
        ResponseClientOutdated.Code = ControlResponseCode.ClientOutdated;

        ResponseInUse  = new ControlResponseHeader();
        ResponseInUse.Code = ControlResponseCode.InUse;
    }

}

File: src/csharp/VRE.Vridge.API.Client/Messages/Control/Responses/ControlResponseHeader.cs
namespace VRE.Vridge.API.Client.Messages.Control.Responses
{
    public class ControlResponseHeader : BaseControlMessage
    {   
        // Predefined responses
        public static ControlResponseHeader ResponseInUse = 
            new ControlResponseHeader(ControlResponseCode.InUse);

        public static ControlResponseHeader ResponseClientOutdated =
            new ControlResponseHeader(ControlResponseCode.ClientOutdated);

        public static ControlResponseHeader ResponseNotAvailable =
            new ControlResponseHeader(ControlResponseCode.NotAvailable);

        public static ControlResponseHeader ResponseServerOutdated =
            new ControlResponseHeader(ControlResponseCode.ServerOutdated);

        public ControlResponseHeader(ControlResponseCode code) : base(3, (int) code)
        {
        }
    }
}
